base/operators.jl

```diff
@@ -961,6 +961,7 @@ const => = Pair
 start(p::Pair) = 1
 done(p::Pair, i) = i>2
 next(p::Pair, i) = (getfield(p,i), i+1)
+eltype{A,B}(p::Pair{A,B}) = Union{A,B}
```

Comment:
why union? if a pair is thought of as one element of a Dict, then it's more like there's a keytype and a valtype and the relationship/combination between the two isn't that meaningful